simulator/guest_operations_manager.go

```diff
@@ -108,4 +115,327 @@ func (m *GuestFileManager) InitiateFileTransferFromGuest(ctx *Context, req *type
+	if fault != nil {
+		body.Fault_ = Fault("", fault)
+		return body
+	}
+
+	body.Res = new(types.DeleteFileInGuestResponse)
+
+	return body
+}
+
+func (m *GuestFileManager) DeleteDirectoryInGuest(ctx *Context, req *types.DeleteDirectoryInGuest) soap.HasFault {
+	body := new(methods.DeleteDirectoryInGuestBody)
+
+	args := []string{"rmdir", req.DirectoryPath}
+	if req.Recursive {
+		args = []string{"rm", "-rf", req.DirectoryPath}
+	}
+
+	vm := ctx.Map.Get(req.Vm).(*VirtualMachine)
+
+	_, fault := vm.run.exec(ctx, vm, req.Auth, args)
+	if fault != nil {
+		body.Fault_ = Fault("", fault)
+		return body
+	}
+
+	body.Res = new(types.DeleteDirectoryInGuestResponse)
+
+	return body
+}
+
+func (m *GuestFileManager) MakeDirectoryInGuest(ctx *Context, req *types.MakeDirectoryInGuest) soap.HasFault {
+	body := new(methods.MakeDirectoryInGuestBody)
+
+	args := []string{"mkdir", req.DirectoryPath}
+	if req.CreateParentDirectories {
+		args = []string{"mkdir", "-p", req.DirectoryPath}
+	}
+
+	vm := ctx.Map.Get(req.Vm).(*VirtualMachine)
+
+	_, fault := vm.run.exec(ctx, vm, req.Auth, args)
+	if fault != nil {
+		body.Fault_ = Fault("", fault)
+		return body
+	}
+
+	body.Res = new(types.MakeDirectoryInGuestResponse)
+
+	return body
+}
+
+func (m *GuestFileManager) MoveFileInGuest(ctx *Context, req *types.MoveFileInGuest) soap.HasFault {
+	body := new(methods.MoveFileInGuestBody)
+
+	args := []string{"mv"}
+	if !req.Overwrite {
+		args = append(args, "-n")
+	}
+	args = append(args, req.SrcFilePath, req.DstFilePath)
+
+	vm := ctx.Map.Get(req.Vm).(*VirtualMachine)
+
+	_, fault := vm.run.exec(ctx, vm, req.Auth, args)
+	if fault != nil {
+		body.Fault_ = Fault("", fault)
+		return body
+	}
+
+	body.Res = new(types.MoveFileInGuestResponse)
+
+	return body
+}
+
+func (m *GuestFileManager) MoveDirectoryInGuest(ctx *Context, req *types.MoveDirectoryInGuest) soap.HasFault {
+	body := new(methods.MoveDirectoryInGuestBody)
+
+	args := []string{"mv", req.SrcDirectoryPath, req.DstDirectoryPath}
+
+	vm := ctx.Map.Get(req.Vm).(*VirtualMachine)
+
+	_, fault := vm.run.exec(ctx, vm, req.Auth, args)
+	if fault != nil {
+		body.Fault_ = Fault("", fault)
+		return body
+	}
+
+	body.Res = new(types.MoveDirectoryInGuestResponse)
+
+	return body
+}
+
+func (m *GuestFileManager) ChangeFileAttributesInGuest(ctx *Context, req *types.ChangeFileAttributesInGuest) soap.HasFault {
+	body := new(methods.ChangeFileAttributesInGuestBody)
+
+	vm := ctx.Map.Get(req.Vm).(*VirtualMachine)
+
+	attr, ok := req.FileAttributes.(*types.GuestPosixFileAttributes)
+	if !ok {
+		body.Fault_ = Fault("", new(types.OperationNotSupportedByGuest))
+		return body
+	}
+
+	if attr.Permissions != 0 {
+		args := []string{"chmod", fmt.Sprintf("%#o", attr.Permissions), req.GuestFilePath}
+
+		_, fault := vm.run.exec(ctx, vm, req.Auth, args)
+		if fault != nil {
+			body.Fault_ = Fault("", fault)
+			return body
+		}
+	}
+
+	if attr.GroupId != nil || attr.OwnerId != nil {
+		// TODO: chown
```

Comment:
Should this generate a fault (not implemented) or log a warning?

Reply:
Added mapping to chmod + chgrp commands.